### DISCUSSION TITLE
Advice for tuning?

### DISCUSSION THREAD

**Discussion by cinjon** [acoular/acoular#405] [Q&A] [answered]
Hi. I am trying to visualize the sound being captured by my minidsp uma16. In the code below, I get videos, but it doesn't make so much sense; I expect the noise bubble to move around to more extremes and be quite small and focused. See an example image below.

What am I doing wrong? And what are good strategies for tuning the params? 
Also, if there is a better way to do this than chunking every .1 or .25s, that would be great to know.

Thanks!

Code: https://gist.github.com/cinjon/a438f96332911dc97368ff3852d1275a

<img width="805" alt="Screenshot 2024-12-22 at 8 45 28 AM" src="https://github.com/user-attachments/assets/fab5cf9f-4699-4a29-afce-a367cd50b880" />

(This is similar to https://github.com/acoular/acoular/discussions/350, but I think different enough that it warranted asking for help.)

**Comment by artpelling** [comment]
Hello @cinjon :)

May I ask what frequency range you are looking at? Judging by the default `kwargs` in your gist, it looks like `(3,375)? Is that Hz?
Also, it is important to know distance between grid and array.

From what I am seeing, your result is pretty much expected. The UMA is a rather small array that works best the upper frequency range (maybe around 4kHz). If you are interested in frequencies below 375Hz, you would need an array with a much larger aperture.

As suggested in #350, you could have a look at the array's point spread function with `spectacoular` to get a feel for the spatial resolution.

**Comment by cinjon** [comment]
Ahh, where do you see that it only works well in the upper frequency ranges? Forgive my ignorance.
I am trying to capture street noise, e.g honking, from 8 floors up. As far as I understand, this should be in the 300-1k range.

However, the specific noise that I'm showing up above was from me speaking in different parts of the room trying to calibrate it.

**Comment by esarradj** [comment, marked as answer]
There is a general law for any imaging system that the angular resolution depends on the ratio of aperture and wavelength (see [here](https://en.wikipedia.org/wiki/Angular_resolution) for explanation referring to light, but valid for sound as well). This means you will preferable need an aperture large compared to wavelength.

Deconvolution methods will improve this (you can easily try with Acoular), but bring other problems such as sensitivity to non-freefield conditions.

**Comment by cinjon** [comment]
@esarradj , thanks for that suggestion. It seems like I'm pretty screwed as the 5.5in x 5.5in of the UMA-16, paired w being 8 floors up, means that I'll have pretty bad spatial localization at all of the frequencies I care about.
